### PR TITLE
Add upgrades to presets

### LINKED ISSUE #921: Make presets upgradeable
Following on https://github.com/OpenZeppelin/cairo-contracts/discussions/878

### PR DESCRIPTION
**Pull request by andrew-fleming**
Fixes #921.

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Added entry to CHANGELOG.md 
- [x] Tried the feature on a public network
  -  ERC20: [0x044d72fd28241658716d7fd30e5893f1c63329b55cd9393a2e487a995d462eff](https://sepolia.starkscan.co/contract/0x044d72fd28241658716d7fd30e5893f1c63329b55cd9393a2e487a995d462eff) (sepolia)
  - ERC721: [0x0743cadcd7cfd9240ee959ef49c0dfa52fbea8a5663e478119efae4fd5cb6afc](https://sepolia.starkscan.co/contract/0x0743cadcd7cfd9240ee959ef49c0dfa52fbea8a5663e478119efae4fd5cb6afc) (sepolia)
  - ERC1155: [0x018c5afd88f24176f5414429c62543d989200414a209088dc468126c456f1ab1](https://sepolia.starkscan.co/contract/0x018c5afd88f24176f5414429c62543d989200414a209088dc468126c456f1ab1) (sepolia)

### CODE REVIEW

File: src/presets/erc721.cairo [hunk 1]
```diff
@@ -3,39 +3,64 @@
...
+    use openzeppelin::access::ownable::OwnableComponent;
     use openzeppelin::introspection::src5::SRC5Component;
     use openzeppelin::token::erc721::ERC721Component;
-    use starknet::ContractAddress;
+    use openzeppelin::upgrades::UpgradeableComponent;
+    use openzeppelin::upgrades::interface::IUpgradeable;
+    use starknet::{ContractAddress, ClassHash};
 
+    component!(path: OwnableComponent, storage: ownable, event: OwnableEvent);
     component!(path: ERC721Component, storage: erc721, event: ERC721Event);
```
Comment (ericnordelo) [inline src/presets/erc721.cairo:21]:
```suggestion
     component!(path: ERC721Component, storage: erc721, event: ERC721Event);
    component!(path: OwnableComponent, storage: ownable, event: OwnableEvent);
```

Didn't we agreed on using the main module component first? If that's the case we should update the other presets as well.

File: src/presets/erc721.cairo [hunk 2]
```diff
@@ -3,39 +3,64 @@
...
     component!(path: SRC5Component, storage: src5, event: SRC5Event);
+    component!(path: UpgradeableComponent, storage: upgradeable, event: UpgradeableEvent);
+
+    // OwnableMixin
```
Comment (ericnordelo) [inline src/presets/erc721.cairo:25]:
```suggestion
    // Ownable Mixin
```

This applies for the rest of the files.

File: src/tests/upgrades/test_upgradeable.cairo [hunk 3]
```diff
@@ -87,13 +87,13 @@ fn test_remove_selector_fails_in_v2() {
...
-fn assert_event_upgraded(class_hash: ClassHash, contract: ContractAddress) {
+fn assert_event_upgraded(contract: ContractAddress, class_hash: ClassHash) {
     let event = utils::pop_log::<UpgradeableComponent::Event>(contract).unwrap();
     let expected = UpgradeableComponent::Event::Upgraded(Upgraded { class_hash });
     assert!(event == expected);
 }
 
-fn assert_only_event_upgraded(class_hash: ClassHash, contract: ContractAddress) {
-    assert_event_upgraded(class_hash, contract);
+fn assert_only_event_upgraded(contract: ContractAddress, class_hash: ClassHash) {
```
Comment (ericnordelo) [inline src/tests/upgrades/test_upgradeable.cairo:96]: 👍🏼

**Review by ericnordelo — COMMENTED** [review]
Looking good! I like the interfaces file inside the presets folder. I Left two small suggestions.

Any reason why this is still a draft?

**Comment by andrew-fleming** [comment]
> Any reason why this is still a draft?

@ericnordelo yeah, I'm deploying them all to sepolia to double check that they function as they should

File: docs/modules/ROOT/pages/utils/_class_hashes.adoc [hunk 4]
```diff
@@ -2,10 +2,10 @@
 :class-hash-cairo-version: https://crates.io/crates/cairo-lang-compiler/2.6.3[cairo 2.6.3]
 
 // Class Hashes
-:Account-class-hash: 0x0450f568a8cb6ea1bcce446355e8a1c2e5852a6b8dc3536f495cdceb62e8a7e2
-:ERC1155-class-hash: 0x0165bfb888913ab8adda83e5d859fd1c283460d51f8f694da00c121b65a6fcc8
-:ERC20-class-hash: 0x06aff5bf77ad88f5ccadc80360e3dcdd12e1a8009160458296b8aac9de2240ff
-:ERC721-class-hash: 0x04cd23b9cf1f3206bef278fe4a4cf6c34e88a7ae20d85884e3d6a1bf8df06693
+:AccountUpgradeable-class-hash: 0x0450f568a8cb6ea1bcce446355e8a1c2e5852a6b8dc3536f495cdceb62e8a7e2
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/utils/_class_hashes.adoc:5]: We need to update the references to these on the presets page.

File: docs/modules/ROOT/pages/api/erc20.adoc [hunk 5]
```diff
@@ -443,13 +443,27 @@ include::../utils/_class_hashes.adoc[]
...
-[[ERC20-constructor]]
-==== `[.contract-item-name]#++constructor++#++(ref self: ContractState, name: ByteArray, symbol: ByteArray, fixed_supply: u256, recipient: ContractAddress)++` [.item-kind]#constructor#
+[[ERC20Upgradeable-constructor]]
+==== `[.contract-item-name]#++constructor++#++(ref self: ContractState, name: ByteArray, symbol: ByteArray, fixed_supply: u256, recipient: ContractAddress, owner: ContractAddress)++` [.item-kind]#constructor#
 
 Sets the `name` and `symbol` and mints `fixed_supply` tokens to `recipient`.
+Assigns `owner` as the contract owner with permissions to upgrade.
+
+[#ERC20Upgradeable-external-functions]
+==== External functions
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/api/erc20.adoc:463]: We are missing this section above.

File: docs/modules/ROOT/pages/api/erc721.adoc [hunk 6]
```diff
@@ -718,15 +718,28 @@ include::../utils/_class_hashes.adoc[]
...
+[[ERC721Upgradeable-constructor]]
+==== `[.contract-item-name]#++constructor++#++(ref self: ContractState, name: ByteArray, symbol: ByteArray, recipient: ContractAddress, token_ids: Span<u256>, base_uri: ByteArray, owner: ContractAddress)++` [.item-kind]#constructor#
 
 Sets the `name` and `symbol`.
-
 Mints `token_ids` tokens to `recipient` and sets the `base_uri`.
+Assigns `owner` as the contract owner with permissions to upgrade.
+
+[#ERC721Upgradeable-external-functions]
+==== External functions
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/api/erc721.adoc:739]: We are missing this section above.

File: docs/modules/ROOT/pages/api/erc1155.adoc [hunk 7]
```diff
@@ -652,20 +652,34 @@ include::../utils/_class_hashes.adoc[]
...
+Assigns `owner` as the contract owner with permissions to upgrade.
 
 Requirements:
 
 - `to` is either an account contract (supporting ISRC6) or
    supports the `IERC1155Receiver` interface.
 - `token_ids` and `values` must have the same length.
+
+[#ERC1155Upgradeable-external-functions]
+==== External functions
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/api/erc1155.adoc:679]: We are missing this section above.

File: docs/modules/ROOT/pages/api/account.adoc [hunk 8]
```diff
@@ -504,27 +504,27 @@ Emitted when a `public_key` is removed.
 == Presets
 
 [.contract]
-[[Account]]
-=== `++Account++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.11.0/src/presets/account.cairo[{github-icon},role=heading-link]
+[[AccountUpgradeable]]
+=== `++AccountUpgradeable++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.11.0/src/presets/account.cairo[{github-icon},role=heading-link]
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/api/account.adoc:508]: There is an example in the test utilities docs that is still showing `presets::Account` (I cannot select it).

File: src/presets/interfaces/account.cairo [hunk 9]
```diff
@@ -0,0 +1,35 @@
+use starknet::account::Call;
+use starknet::{ContractAddress, ClassHash};
+
+#[starknet::interface]
+trait IAccountUpgradeable<TState> {
```
Comment (ericnordelo) [inline src/presets/interfaces/account.cairo:5]:
```suggestion
trait AccountUpgradeableABI<TState> {
```

I think this is more accurate and consistent, wdyt? If we agree we need to update the rest as well.

**Review by ericnordelo — COMMENTED** [review]
Looking good, left some small suggestions.

Let's not forget to update the KNOWN_ORDER names in the script that updates the class hashes

**Review by ericnordelo — APPROVED** [review]
LGTM!

**Pull request merged**